addons/account/tests/test_account_move_send.py

```diff
@@ -1107,3 +1107,18 @@ def test_get_sending_settings(self):
         }
         results = wizard._get_sending_settings()
         self.assertDictEqual(results, expected_results)
+
+    def test_email_subtitle(self):
+        """ Test email notification subtitle for partner with and without name. """
+        invoice1 = self.init_invoice("out_invoice", partner=self.partner_a, amounts=[1000], post=True)
+        context1 = invoice1._notify_by_email_prepare_rendering_context(message=self.env['mail.message'])
+        subtitles1 = context1.get('subtitles')
+        self.assertTrue(subtitles1)
+        self.assertEqual(subtitles1[0], 'INV/2019/00001 - partner_a')
+
+        partner_no_name = self.env['res.partner'].create({'type': 'invoice', 'parent_id': self.partner_a.id})
+        invoice2 = self.init_invoice("out_invoice", amounts=[1000], partner=partner_no_name, post=True)
+        context2 = invoice2._notify_by_email_prepare_rendering_context(message=self.env['mail.message'])
+        subtitles2 = context2.get('subtitles')
+        self.assertNotIn('False', subtitles2[0])
+        self.assertEqual(subtitles2[0], 'INV/2019/00002')
```

Comment:
need to optimise the code, no need for extra variables to assign value and extra check for `subtitles`. 
use one invoice to check both cases.